### PR TITLE
Pay 3900 - Org id Integration(Pay - 3899,3900,3901)

### PR DESCRIPTION
**Pull request by vigneshpushkaranHMCTS**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/CreditAccountPaymentController.java [hunk 1]
```diff
@@ -84,26 +95,41 @@ public CreditAccountPaymentController(@Qualifier("loggingCreditAccountPaymentSer
...
     })
     @PostMapping(value = "/credit-account-payments")
     @ResponseBody
     @Transactional
-    public ResponseEntity<PaymentDto> createCreditAccountPayment(@Valid @RequestBody CreditAccountPaymentRequest creditAccountPaymentRequest) throws CheckDigitException {
+    public ResponseEntity<PaymentDto> createCreditAccountPayment(@Valid @RequestBody CreditAccountPaymentRequest creditAccountPaymentRequest, @RequestHeader(required = false) MultiValueMap<String, String> headers) throws CheckDigitException {
         String paymentGroupReference = PaymentReference.getInstance().getNext();
 
+        LOG.info("PBA Old Config Service Names : {}", pbaConfig1ServiceNames);
+        Boolean isPBAConfig1Journey = pbaConfig1ServiceNames.contains(creditAccountPaymentRequest.getService())
```
Comment (AyanDuttaHMCTS) [inline api/src/main/java/uk/gov/hmcts/payment/api/controllers/CreditAccountPaymentController.java:109]: Comment to be added to remove once Services are on boarded to PBA2

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/PaymentRecordController.java [hunk 2]
```diff
@@ -93,7 +97,7 @@ public PaymentRecordController(PaymentRecordService<PaymentFeeLink, String> paym
             .currency(paymentRecordRequest.getCurrency().getCode())
             .paymentProvider(paymentProvider)
             .externalReference(paymentRecordRequest.getExternalReference())
-            .serviceType(paymentRecordRequest.getService().getName())
+            .serviceType(paymentService.getServiceNameByCode(paymentRecordRequest.getService()))
```
Comment (AyanDuttaHMCTS) [inline api/src/main/java/uk/gov/hmcts/payment/api/controllers/PaymentRecordController.java:100]: Comment to add for Digital Bar

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/RemissionController.java [hunk 3]
```diff
@@ -42,6 +39,12 @@
     @Autowired
     private RemissionDtoMapper remissionDtoMapper;
 
+    @Autowired
```
Comment (AyanDuttaHMCTS) [inline api/src/main/java/uk/gov/hmcts/payment/api/controllers/RemissionController.java:42]: to be removed

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/RemissionController.java [hunk 4]
```diff
@@ -53,9 +56,11 @@
     @PostMapping(value = "/remission")
     @ResponseBody
     @Deprecated
```
Comment (AyanDuttaHMCTS) [inline api/src/main/java/uk/gov/hmcts/payment/api/controllers/RemissionController.java:58]: If not being used, clean up required

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/RemissionController.java [hunk 5]
```diff
@@ -77,9 +82,11 @@
     })
     @PostMapping(value = "/remissions")
     @ResponseBody
-    public ResponseEntity<RemissionDto> createRemission(@Valid @RequestBody RemissionRequest remissionRequest)
+    public ResponseEntity<RemissionDto> createRemission(@Valid @RequestBody RemissionRequest remissionRequest,
+                                                        @RequestHeader(required = false) MultiValueMap<String, String> headers)
         throws CheckDigitException {
-        remissionValidator.validate(remissionRequest);
```
Comment (AyanDuttaHMCTS) [inline api/src/main/java/uk/gov/hmcts/payment/api/controllers/RemissionController.java:82]: remissionValidator should be removed as unused

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/RemissionController.java [hunk 6]
```diff
@@ -101,15 +108,23 @@
     public ResponseEntity<RemissionDto> createRetrospectiveRemission(
         @PathVariable("payment-group-reference") String paymentGroupReference,
         @PathVariable("unique_fee_id") Integer feeId,
+        @RequestHeader(required = false) MultiValueMap<String, String> headers,
         @Valid @RequestBody RemissionRequest remissionRequest) throws CheckDigitException {
-        remissionValidator.validate(remissionRequest);
+
+        getOrganisationalDetails(headers, remissionRequest);
+        LOG.info("SiteId : {} ", remissionRequest.getSiteId());
```
Comment (vigneshpushkaranHMCTS) [inline api/src/main/java/uk/gov/hmcts/payment/api/controllers/RemissionController.java:115]: Remove log

File: api/src/main/java/uk/gov/hmcts/payment/api/dto/BulkScanPaymentRequestStrategic.java [hunk 7]
```diff
@@ -66,10 +63,12 @@
     @NotNull
     private String bankedDate;
 
-    @NotEmpty
     @JsonProperty("site_id")
```
Comment (vigneshpushkaranHMCTS) [inline api/src/main/java/uk/gov/hmcts/payment/api/dto/BulkScanPaymentRequestStrategic.java:66]: rid of site id same as telephony payment

File: api/src/main/java/uk/gov/hmcts/payment/api/dto/RemissionRequest.java [hunk 8]
```diff
@@ -42,9 +43,11 @@
     @Valid
     private FeeDto fee;
 
-    @NotNull
     private String siteId;
```
Comment (vigneshpushkaranHMCTS) [inline api/src/main/java/uk/gov/hmcts/payment/api/dto/RemissionRequest.java:46]: rid of site id same telephony payment also check for DTO **usage** across application.

File: api/src/main/resources/application-local.properties [hunk 9]
```diff
@@ -94,8 +94,10 @@ launchDarkly.flushInterval=10
 bulk.scanning.payments.processed.url=http://localhost:8004
 idam.s2s-auth.totp_secret=${PAYMENT_S2S_SECRET:AAAAAAAAAAAAAAAA}
 idam.s2s-auth.microservice=${PAYMENT_S2S_MICROSERVICE_NAME:internal}
+pba.config1.service.names=${PBA_CONFIG1_SERVICE_NAMES:PROBATE,CMC}
```
Comment (AyanDuttaHMCTS) [inline api/src/main/resources/application-local.properties:97]: Remove Local Configs

File: api/src/main/resources/application.properties [hunk 10]
```diff
@@ -148,5 +148,7 @@ launchDarkly.socketTimeout=3
 launchDarkly.flushInterval=10
 pba.config1.service.names=${PBA_CONFIG1_SERVICE_NAMES:dummy}
 
+
+rd.location.url=${RD_LOCATION_BASE_URL:http://rd-location-ref-api-aat.service.core-compute-aat.internal}
```
Comment (AyanDuttaHMCTS) [inline api/src/main/resources/application.properties:152]: Sync with Other API configs as "http://dummy.dummy.net"

File: charts/payment-api/values.preview.template.yaml [hunk 11]
```diff
@@ -18,8 +18,17 @@ java:
     #Below key required for preview pipeline
     GOV_PAY_AUTH_KEY_CMC: 8ln764q6jrbgqvvvv40uggsea3h1p602v79gdrt7ni66mrmpk416ulohvd
     LAUNCH_DARKLY_SDK_KEY: sdk-06456c8b-312c-475c-b154-ca585f5ca063
-    CCPAY_BULK_SCANNING_API_URL: http://ccpay-bulkscanning-api-aat.service.core-compute-aat.internal
+    CCPAY_BULK_SCANNING_API_URL: http://ccpay-bulkscanning-api-pr-285.service.core-compute-preview.internal/
     PAYMENT_S2S_SECRET: ETNY5CBN7NR62T6M
+    card.payments.email.to: vignesh.pushkaran@hmcts.net
```
Comment (AyanDuttaHMCTS) [inline charts/payment-api/values.preview.template.yaml:23]: Change to QA's Email

File: model/src/main/java/uk/gov/hmcts/payment/api/service/ReferenceDataServiceImpl.java [hunk 12]
```diff
@@ -1,21 +1,90 @@
...
+
     @Override
     public List<SiteDTO> getSiteIDs() {
         return SiteDTO.fromSiteList(siteService.getAllSites());
     }
+
+    @Override
+    public OrganisationalServiceDto getOrganisationalDetail(String caseType, MultiValueMap<String, String> headers) {
+
+        MultiValueMap<String, String> headerMultiValueMapForOrganisationalDetail = new LinkedMultiValueMap<String, String>();
```
Comment (AyanDuttaHMCTS) [inline model/src/main/java/uk/gov/hmcts/payment/api/service/ReferenceDataServiceImpl.java:59]: Optimise code

File: api/src/main/resources/application-local.properties (hunk 9 above)
Comment (vigneshpushkaranHMCTS) [reply api/src/main/resources/application-local.properties]: removed

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/RemissionController.java (hunk 3 above)
Comment (Jennings-HMCTS) [reply api/src/main/java/uk/gov/hmcts/payment/api/controllers/RemissionController.java:37]: referencedataservice is used

**Review by YasharRahvar — APPROVED** [review]

**Review by krishnanuthalapati — APPROVED** [review]

**Pull request merged**